From bb33080bd344015d07fd0fde682abbdbc4573d1d Mon Sep 17 00:00:00 2001
From: Yiqun Lin <yqlin@apache.org>
Date: Sat, 4 Mar 2017 13:41:49 +0800
Subject: [PATCH 2763/2848] HDFS-11494. Log message when DN is not selected
 for block replication. Contributed by Yiqun Lin.

(cherry picked from commit 2288857e3bf8c738d6c1ca7982523d6e4788f859)

Change-Id: If7e38dc9723ce6bf3e4fb887b21b2eda64873c1e
---
 .../server/blockmanagement/DatanodeDescriptor.java |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeDescriptor.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeDescriptor.java
index 1ffdd09..a411ef5 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeDescriptor.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeDescriptor.java
@@ -640,6 +640,10 @@ public DatanodeStorageInfo chooseStorage4Block(StorageType t,
       }
     }
     if (requiredSize > remaining - scheduledSize) {
+      LOG.debug(
+          "The node {} does not have enough {} space (required={},"
+          + " scheduled={}, remaining={}).",
+          this, t, requiredSize, scheduledSize, remaining);
       return null;
     }
     return storage;
-- 
1.7.9.5

